test/strscan/test_stringscanner.rb

```diff
@@ -946,6 +946,16 @@ def test_scan_integer_encoding
     end
   end
 
+  def test_scan_integer_matched 
```

Comment:
You need `omit("scan_integer isn't implemented on TruffleRuby yet") if RUBY_ENGINE == "truffleruby"` as well here (will address those soon, TruffleRuby is implementing the new method)